### DISCUSSION TITLE
Eliminate per page index.pageContext.json files

### DISCUSSION THREAD

**Discussion by kripainc** [vikejs/vike#649] [Discussion]
Is there in configuration option to suppress generating index.pageContext.json files?
(content of this file is page specific and looks like:  {"pageContext":{"_pageId":"/pages/select/@id"}}

I am migrating a large site (90K+ pages) from Gatsby/React to vite-plugin-ssr. This creates 90K+ directories with a single 'index.pageContext.json' file. It's also fetched by client when rendering the page.

ENVIRONMENT: vite: 4.0.4, vite-plugin-ssr: 0.4.82, node: v16.18.0 (linux Kubuntu), 

Also, using vite config option to suppress building sub-directories for html files (which is working correctly):  ssr({ prerender: {  noExtraDir: true, }}), and did not specify 'export { passToClient }' option in any of my *.page.server.tsx files.
All pages are SSG, and do not require any data to be injected on the client side.

Q1: Is there a way to eliminate generating sub-folders for each index.pageContext.json file?
Q2: Is there a way to suppress fetching individual index.pageContext.json file?

Thanks for the great vite plugin. I really enjoyed the DX.

**Comment by brillout** [comment]
If you don't define any `onBeforeRender()` then these `.pageContext.json` files won't ever be used/requested. You're right that VPS could be smart enough to simply not generate them.

Is it causing any problems?

**Reply by kripainc** [reply]
onBeforeRender() is required (for development/testing) as it's used for fetching data from DB.
However, just for testing, I did comment out the onBeforeRender() and the .pageContext.json files are still generated in it's own folder but NOT fetched by client.  Thank you for the tip.

**Reply by brillout** [reply]
Exactly. Let me know if it causes any disturbances and I’ll fix it. Thanks for pointing this out.

**Reply by brillout** [reply]
Also FYI the [V1 design](https://github.com/brillout/vite-plugin-ssr/issues/578) enables quite some powerful configuration and I can see that you'll even be able to conditionally define `onBeforeRender()` depending on dev/prod, let's see.
